spec/rubocop/cop/naming/memoized_instance_variable_name_spec.rb

```diff
@@ -194,6 +200,83 @@ def initialize
           end
         end
       end
+
+      context 'with dynamically defined methods' do
+        context 'when the variable name matches the method name' do
+          it 'does not register an offense' do
+            expect_no_offenses(<<~RUBY)
+              define_method(:values) do
+                @values ||= do_something
+              end
+            RUBY
+          end
+        end
+
+        context 'when the variable name does not match the method name' do
+          it 'registers an offense' do
+            expect_offense(<<~RUBY)
+              define_method(:values) do
+                @foo ||= do_something
+                ^^^^ Memoized variable `@foo` does not match method name `values`. Use `@values` instead.
+              end
+            RUBY
+          end
+        end
+
+        context 'when a method is defined inside a module callback' do
+          context 'when the method matches' do
+            it 'does not register an offense' do
+              expect_no_offenses(<<~RUBY)
+                def self.inherited(klass)
+                  klass.define_method(:values) do
+                    @values ||= do_something
+                  end
+                end
+              RUBY
+            end
+          end
+
+          context 'when the method does not match' do
+            it 'it registers an offense' do
```

Comment:
Can you update this one?

```suggestion
            it 'registers an offense' do
```

Reply:
🤦

Reply:
Fixed!